### ISSUE TITLE
JDK 9 module-info.java handling

### ISSUE THREAD

**Issue by sormuras** [google/google-java-format#75]
Having a `module-info.java` in the source set leads to unexpected format exceptions starting with

```
src/main/java/module-info.java:1:1: error: Syntax error on token "module", package expected
```

Solution 1

Skip `module-info.java`  although it ends with `.java` Line `Main.java`:[118](https://github.com/google/google-java-format/blob/master/core/src/main/java/com/google/googlejavaformat/java/Main.java#L118)  already skips non-Java files. Could easily be handled here.

Solution 2

Support `module-info.java` [syntax](http://cr.openjdk.java.net/~mr/jigsaw/spec/lang-vm.html) and reformat it as well.

**Comment by cushon** [comment]
I'd prefer (2), but I'm also not sure what the timeline is for ecj's java 9 support.

**Comment by sormuras** [comment]
I prefer (2) as well, but (1) could serve as an interim solution, until ECJ supports Java 9.

**Comment by sormuras** [comment]
Latest module syntax with layout example can be found here: http://cr.openjdk.java.net/~mr/jigsaw/spec/lang-vm.html

**Comment by sormuras** [comment]
Any news on ecj's java 9 support? The `com.google.errorprone:javac:9-dev-r3297-1-shaded` does not contain it, I guess.

**Comment by cushon** [comment]
We migrated from ecj to javac. javac has JDK 9 support, but I'm not sure if the version the formatter is using is new enough.

**Comment by sormuras** [comment]
According to http://central.maven.org/maven2/com/google/errorprone/javac/ there's a:
- [9-dev-r4023-2](http://central.maven.org/maven2/com/google/errorprone/javac/9-dev-r4023-2/)

But it is missing the `-shaded` suffix: [9-dev-r3297-1-shaded](http://central.maven.org/maven2/com/google/errorprone/javac/9-dev-r3297-1-shaded/)

Although the file size suggests it _does_ include all dependencies...

**Comment by sormuras** [comment]
Seems like the update of `javac` would solve this problem, if `-source 9` could be specified:

```
com.google.googlejavaformat.java.FormatterException: 1:2: error: modules are not supported in -source 1.8
  (use -source 9 or higher to enable modules)

	at com.google.googlejavaformat.java.Formatter.getFormatReplacements(Formatter.java:242)
	at com.google.googlejavaformat.java.Formatter.formatSource(Formatter.java:216)
	at com.google.googlejavaformat.java.Formatter.formatSource(Formatter.java:202)
	at com.google.googlejavaformat.java.Java9SourceTest.issue75(Java9SourceTest.java:30)
```

**Comment by sormuras** [comment]
On may way!

**Comment by sormuras** [comment]
LGTM.

All branches of the [JLS module production](http://cr.openjdk.java.net/~mr/jigsaw/spec/java-se-9-jls-diffs.pdf) are hit by the example `M1` input/output pair:
```
ModularCompilationUnit:
  {ImportDeclaration} ModuleDeclaration

ModuleDeclaration:
  {Annotation} [open] module Identifier {. Identifier}
  { {ModuleDirective} }

ModuleDirective:
  requires {RequiresModifier} ModuleName ;
  exports PackageName [to ModuleName {, ModuleName}] ;
  opens PackageName [to ModuleName {, ModuleName}] ;
  uses TypeName ;
  provides TypeName with TypeName {, TypeName} ;

RequiresModifier:
  (one of)
  transitive static
```

**Comment by sormuras** [comment]
> I'll cut a new release soon that includes that change unless any issues turn up.

Great news! _(With tiny chance seeing #106 included as well?)_

**Comment by sormuras** [comment]
Also all 30 `module-info.java` files in https://github.com/AdoptOpenJDK/jdk9-jigsaw were formatted successfully.

<details>
 <summary>Generated patch looks ok</summary>

```
Index: session-2-jlink/02_JMod/src/org.astro/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-2-jlink/02_JMod/src/org.astro/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-2-jlink/02_JMod/src/org.astro/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module org.astro { 
-	exports org.astro;
-}
\ No newline at end of file
+module org.astro {
+  exports org.astro;
+}
Index: session-2-jlink/01_JLink/src/org.astro/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-2-jlink/01_JLink/src/org.astro/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-2-jlink/01_JLink/src/org.astro/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module org.astro { 
-	exports org.astro;
-}
\ No newline at end of file
+module org.astro {
+  exports org.astro;
+}
Index: session-2-jlink/02_JMod/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-2-jlink/02_JMod/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-2-jlink/02_JMod/src/com.greetings/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module com.greetings { 
-	requires org.astro;
-}
\ No newline at end of file
+module com.greetings {
+  requires org.astro;
+}
Index: session-2-jlink/01_JLink/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-2-jlink/01_JLink/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-2-jlink/01_JLink/src/com.greetings/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module com.greetings { 
-	requires org.astro;
-}
\ No newline at end of file
+module com.greetings {
+  requires org.astro;
+}
Index: session-1-jigsaw-intro/06_Services/src/com.socket/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/06_Services/src/com.socket/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/06_Services/src/com.socket/module-info.java	(revision )
@@ -1,5 +1,6 @@
 module com.socket {
-	exports com.socket;
-	exports com.socket.spi;
-	uses com.socket.spi.NetworkSocketProvider;
-}
\ No newline at end of file
+  exports com.socket;
+  exports com.socket.spi;
+
+  uses com.socket.spi.NetworkSocketProvider;
+}
Index: session-1-jigsaw-intro/04_Packaging/src/org.astro/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/04_Packaging/src/org.astro/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/04_Packaging/src/org.astro/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module org.astro { 
-	exports org.astro;
-}
\ No newline at end of file
+module org.astro {
+  exports org.astro;
+}
Index: session-1-jigsaw-intro/06_Services/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/06_Services/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/06_Services/src/com.greetings/module-info.java	(revision )
@@ -1,3 +1,3 @@
 module com.greetings {
-	requires com.socket;
-}
\ No newline at end of file
+  requires com.socket;
+}
Index: session-1-jigsaw-intro/04_Packaging/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/04_Packaging/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/04_Packaging/src/com.greetings/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module com.greetings { 
-	requires org.astro;
-}
\ No newline at end of file
+module com.greetings {
+  requires org.astro;
+}
Index: session-1-jigsaw-intro/01_Greetings/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/01_Greetings/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/01_Greetings/src/com.greetings/module-info.java	(revision )
@@ -1,1 +1,1 @@
-module com.greetings { }
\ No newline at end of file
+module com.greetings {}
Index: session-1-jigsaw-intro/06_Services/src/org.fastsocket/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/06_Services/src/org.fastsocket/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/06_Services/src/org.fastsocket/module-info.java	(revision )
@@ -1,5 +1,6 @@
 module org.fastsocket {
-	requires com.socket;
-	provides com.socket.spi.NetworkSocketProvider
-		with org.fastsocket.FastNetworkSocketProvider;
-}
\ No newline at end of file
+  requires com.socket;
+
+  provides com.socket.spi.NetworkSocketProvider with
+      org.fastsocket.FastNetworkSocketProvider;
+}
Index: session-1-jigsaw-intro/02_GreetingsWorld/src/org.astro/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/02_GreetingsWorld/src/org.astro/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/02_GreetingsWorld/src/org.astro/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module org.astro { 
-	exports org.astro;
-}
\ No newline at end of file
+module org.astro {
+  exports org.astro;
+}
Index: session-1-jigsaw-intro/09_Automodules/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/09_Automodules/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/09_Automodules/src/com.greetings/module-info.java	(revision )
@@ -1,4 +1,6 @@
 module com.greetings {
-    requires junit;
-    exports com.greetings to junit;
-}
\ No newline at end of file
+  requires junit;
+
+  exports com.greetings to
+      junit;
+}
Index: session-1-jigsaw-intro/05_Missing_exports/src/org.astro/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/05_Missing_exports/src/org.astro/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/05_Missing_exports/src/org.astro/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module org.astro { 
-	// exports org.astro;
-}
\ No newline at end of file
+module org.astro {
+  // exports org.astro;
+}
Index: session-1-jigsaw-intro/05_Missing_requires/src/org.astro/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/05_Missing_requires/src/org.astro/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/05_Missing_requires/src/org.astro/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module org.astro { 
-	exports org.astro;
-}
\ No newline at end of file
+module org.astro {
+  exports org.astro;
+}
Index: session-1-jigsaw-intro/02_GreetingsWorld/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/02_GreetingsWorld/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/02_GreetingsWorld/src/com.greetings/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module com.greetings { 
-	requires org.astro;
-}
\ No newline at end of file
+module com.greetings {
+  requires org.astro;
+}
Index: session-1-jigsaw-intro/05_Missing_exports/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/05_Missing_exports/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/05_Missing_exports/src/com.greetings/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module com.greetings { 
-	 requires org.astro;
-}
\ No newline at end of file
+module com.greetings {
+  requires org.astro;
+}
Index: session-1-jigsaw-intro/05_Missing_requires/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/05_Missing_requires/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/05_Missing_requires/src/com.greetings/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module com.greetings { 
-	// requires org.astro;
-}
\ No newline at end of file
+module com.greetings {
+  // requires org.astro;
+}
Index: session-1-jigsaw-intro/08_ModulesExportConflict/src/org.astro/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/08_ModulesExportConflict/src/org.astro/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/08_ModulesExportConflict/src/org.astro/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module org.astro { 
-	exports org.astro;
-}
\ No newline at end of file
+module org.astro {
+  exports org.astro;
+}
Index: session-1-jigsaw-intro/03_MultiModuleCompilation/src/org.astro/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/03_MultiModuleCompilation/src/org.astro/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/03_MultiModuleCompilation/src/org.astro/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module org.astro { 
-	exports org.astro;
-}
\ No newline at end of file
+module org.astro {
+  exports org.astro;
+}
Index: session-1-jigsaw-intro/08_ModulesExportConflict/src/org.astro2/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/08_ModulesExportConflict/src/org.astro2/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/08_ModulesExportConflict/src/org.astro2/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module org.astro { 
-	exports org.astro;
-}
\ No newline at end of file
+module org.astro {
+  exports org.astro;
+}
Index: session-1-jigsaw-intro/07_patch_module_option/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/07_patch_module_option/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/07_patch_module_option/src/com.greetings/module-info.java	(revision )
@@ -1,2 +1,1 @@
-module com.greetings {
-}
\ No newline at end of file
+module com.greetings {}
Index: session-1-jigsaw-intro/08_ModulesExportConflict/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/08_ModulesExportConflict/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/08_ModulesExportConflict/src/com.greetings/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module com.greetings { 
-	requires org.astro;
-}
\ No newline at end of file
+module com.greetings {
+  requires org.astro;
+}
Index: session-1-jigsaw-intro/03_MultiModuleCompilation/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/03_MultiModuleCompilation/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/03_MultiModuleCompilation/src/com.greetings/module-info.java	(revision )
@@ -1,3 +1,3 @@
-module com.greetings { 
-	requires org.astro;
-}
\ No newline at end of file
+module com.greetings {
+  requires org.astro;
+}
Index: session-1-jigsaw-intro/07_patch_ConcurrentHashMap/src/com.greetings/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-1-jigsaw-intro/07_patch_ConcurrentHashMap/src/com.greetings/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-1-jigsaw-intro/07_patch_ConcurrentHashMap/src/com.greetings/module-info.java	(revision )
@@ -1,2 +1,1 @@
-module com.greetings {
-}
\ No newline at end of file
+module com.greetings {}
Index: session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.rest/src/main/java/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.rest/src/main/java/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.rest/src/main/java/module-info.java	(revision )
@@ -1,13 +1,11 @@
 module monitor.rest {
-	requires java.xml.bind;
-	requires java.xml.ws.annotation;
-
-	requires monitor.utils;
-	requires monitor.statistics;
-
-	requires spark.core;
-	requires jackson.core;
-	requires jackson.databind;
+  requires java.xml.bind;
+  requires java.xml.ws.annotation;
+  requires monitor.utils;
+  requires monitor.statistics;
+  requires spark.core;
+  requires jackson.core;
+  requires jackson.databind;
 
-	exports monitor.rest;
+  exports monitor.rest;
 }
Index: session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.utils/src/main/java/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.utils/src/main/java/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.utils/src/main/java/module-info.java	(revision )
@@ -1,4 +1,4 @@
 module monitor.utils {
-	// requires
-	exports monitor.utils;
+  // requires
+  exports monitor.utils;
 }
Index: session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.observer/src/main/java/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.observer/src/main/java/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.observer/src/main/java/module-info.java	(revision )
@@ -1,3 +1,3 @@
 module monitor.observer {
-	exports monitor.observer;
+  exports monitor.observer;
 }
Index: session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.statistics/src/main/java/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.statistics/src/main/java/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.statistics/src/main/java/module-info.java	(revision )
@@ -1,4 +1,5 @@
 module monitor.statistics {
-	requires monitor.observer;
-	exports monitor.statistics;
+  requires monitor.observer;
+
+  exports monitor.statistics;
 }
Index: session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.observer.beta/src/main/java/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.observer.beta/src/main/java/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.observer.beta/src/main/java/module-info.java	(revision )
@@ -1,4 +1,5 @@
 module monitor.observer.beta {
-	requires monitor.observer;
-	exports monitor.observer.beta;
+  requires monitor.observer;
+
+  exports monitor.observer.beta;
 }
Index: session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.observer.alpha/src/main/java/module-info.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.observer.alpha/src/main/java/module-info.java	(revision 46cb9e190b44d0d8aa4f2f4544ebc41251939c52)
+++ session-3-refactoring-migration/04_ServiceMonitor_modularisation_to_java_9/monitor.observer.alpha/src/main/java/module-info.java	(revision )
@@ -1,4 +1,5 @@
 module monitor.observer.alpha {
-	requires monitor.observer;
-	exports monitor.observer.alpha;
+  requires monitor.observer;
+
+  exports monitor.observer.alpha;
 }
```
</details>

**Comment by sormuras** [comment]
While running against all 350 module descriptors found in openjdk, i found something in `AdoptOpenJDK\openjdk-jdk9\langtools\test\jdk\javadoc\doclet\testModules\moduletags\module-info.java`:

```
@Deprecated
module moduletags {
    requires transitive static moduleA;

    exports testpkgmdltags;
}
```

Looks a like file to test the "(one of)" rule from the `requires` production.

**Comment by sormuras** [comment]
See OpenJDK diff/patch here: https://gist.github.com/sormuras/d85612b0e20bd98746aed1b347d064bd

**Comment by cushon** [comment]
Thanks for testing! f74657cbbe11c51f8b2b4810daa7c84ca6c6a552 fixes the issue with `requires transitive static`.

**Comment by sormuras** [comment]
Fixed in release 1.4
